synapse/storage/databases/main/registration.py

```diff
@@ -1915,6 +1940,75 @@ def set_user_deactivated_status_txn(
         self._invalidate_cache_and_stream(txn, self.get_user_by_id, (user_id,))
         txn.call_after(self.is_guest.invalidate, (user_id,))
 
+    async def _background_update_set_approved_flag(
+        self, progress: JsonDict, batch_size: int
+    ) -> int:
+        """Set the 'approved' flag for all already existing users. We want to set it to
```

Comment:
Using `NULL` as approved is an interesting idea, though I'm a bit on the fence with this because imo it's very easy to read `NULL` on a boolean column as a false value, and it also means it needs a bit of special casing when handling it through the admin API (because then we need to explicitly make it `true`). I'd rather not use a default value since, as you point out, it involves rewriting the table in Postgres 10. That version reaching EOL in a couple of months is good news but I'd rather not block this PR on that.

It's also worth noting that this background update is basically a copy of the one we already have for setting the deactivation flag.

Reply:
Actually since writing this PR a similar question came up in https://github.com/matrix-org/synapse/pull/13799, to which the decision was to treat `NULL` as `true`, which isn't _that_ much of a faff in the end. So let's go with that.